File: .config/i3blocks/scripts/brightness.py
#!/bin/python
# Custom i3blocks python script for displaying current screen brightness
#
# Those are the brightness paths on my laptop, change it properly
######################################################################
import math
brighPath = '/sys/class/backlight/intel_backlight/actual_brightness'
maxBrighPath = '/sys/class/backlight/intel_backlight/max_brightness'
######################################################################
file = open(brighPath)
brightness = int(file.read())
file.close()
file = open(maxBrighPath)
maxBrightness = int(file.read())
file.close()
brightnessPerc = math.ceil(brightness * 100 / maxBrightness)

icon = "☀️"

print(' <span color="orange">{}{:>1}%</span> '.format(icon, brightnessPerc))

File: .config/i3blocks/scripts/brightness/brightness.zig
const std = @import("std");

const stdout = std.io.getStdOut().writer();

pub fn main() !void {
    const brighPath = "/sys/class/backlight/intel_backlight/actual_brightness";
    const maxBrighPath = "/sys/class/backlight/intel_backlight/max_brightness";

    const brightness = try readFile(brighPath) * 100;
    const maxBrightness = try readFile(maxBrighPath);

    const brightness_perc = brightness / maxBrightness;

    try stdout.print(" <span color='orange'>☀️{:>1}%</span> \n ", .{brightness_perc});
}

fn readFile(path: []const u8) !u64 {
    var file = try std.fs.openFileAbsolute(path, .{});
    defer file.close();

    var buffer: [16]u8 = undefined;
    const bytes_read = try file.readAll(&buffer);
    const content = buffer[0..bytes_read];

    const trimmed = std.mem.trim(u8, content, "\n");
    return std.fmt.parseInt(u64, trimmed, 10);
}
